Jetchat/app/src/main/java/com/example/compose/jetchat/JetchatApplication.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.compose.jetchat
+
+import android.app.Application
+import com.example.compose.jetchat.data.AppContainer
+import com.example.compose.jetchat.data.AppContainerImpl
+
+class JetchatApplication : Application() {
+
+    // AppContainer instance used by the rest of classes to obtain dependencies
+    lateinit var container: AppContainer
+
+    override fun onCreate() {
+        super.onCreate()
+        container = AppContainerImpl()
+    }
```

Comment:
I suggest remove the ceremony with the `AppContainer` and the `MessageRepository`. It doesn't represent a real-world use-case anyway and might actually just be confusing for someone trying to understand the UI.

I suggest creating `val unreadMessages = initialMessages.filter { it.author != "me" }` in the `FakeData.kt` and just read it in the glance widget.

This sample represents UI sample and we don't need to make it more complicated with DI pattern. There are other samples for that 👍🏼